### ISSUE TITLE
Expose workflow structure and analysis results in richer JSON reports

### ISSUE THREAD

**Issue by kjanat** [kjanat/actionlint#189]
The current JSON check output contains only diagnostics:

```json
{"schema_version":1,"diagnostics":[]}
```
A clean result gives no inventory of the files checked, their workflow structure, or the analysis performed. actionlint already parses workflows and action metadata and derives semantic information; this information should be available to report consumers.

Expose a versioned analysis report built from the existing parser and analyzers. Retain their results through the shared analysis pipeline so the CLI and GitHub Action can consume the same report.

Desired coverage:

- Files, including clean workflows and analyzed local composite actions, with their analysis status and consumed configuration/dependencies.
- Workflow, job and step structure: identifiers, names, source locations, triggers, `needs`, `uses`, reusable-workflow calls and nested composite invocations.
- Declared values and information already derived by analysis, such as effective shell/working-directory and expression types or statically known values. Preserve unresolved expressions and distinguish unavailable runtime values from analysis that was skipped or failed.
- Diagnostics attributable to files, jobs and steps, with accumulated counts at each level. Preserve workflow-level findings and distinguish an action's source location from its invocation sites.
- Analysis completeness and analyzer coverage, including disabled, unavailable or failed external tools. Empty diagnostics alone should not be the entire account of what happened.

Design and validation:

- Reuse the existing AST, metadata and semantic results; collect them during the existing analysis pass.
- Define a stable public report schema with deterministic ordering and documented source-position/identity conventions. Preserve existing CLI JSON and Action output contracts, with an explicit versioning/opt-in strategy for incompatible changes.
- Decide the file collection shape and diagnostic ownership/reference model. Provide per-level access and rollups without inconsistent duplicate findings.
- Cover clean files, findings at different scopes, parse failures, incomplete analysis, disabled tools, nested/repeated composite calls and expressions whose values require runtime context.

Independent follow-up work. This does not expand the scope of the current pull requests.
